frontend/lib/geo-autocomplete.tsx

```diff
@@ -244,8 +244,8 @@ export class GeoAutocomplete extends React.Component<GeoAutocompleteProps, GeoAu
   render() {
     return (
       <GeoDownshift
-        onChange={this.props.onChange}
-        defaultSelectedItem={this.props.initialValue}
+        onChange={item => item && this.props.onChange(item)}
```

Comment:
It looks like `onChange` has changed to also notify clients when _no_ autocomplete item has been selected, e.g. when the user pressed <kbd>ESC</kbd> while autocompletions were showing.  Because this `<GeoAutocomplete>` component never notified its clients of such events, I simply changed its code to do nothing when it receives such events from downshift, to preserve our original behavior.

An example of this in practice: right now if someone is on the DDO results page and starts changing the search text and presses <kbd>ESC</kbd>, the text field is cleared but the form isn't actually submitted until the user moves focus away from the search field.  If `<GeoAutocomplete>` notifies its parent of the change, however, then pressing <kbd>ESC</kbd> would actually submit the form with an empty search value, which would actually (potentially jarringly) take the user back to the DDO page without any search results.

Reply:
huh, interesting